### PR TITLE
models rework

### PR DESCRIPTION
**Pull request by KMosc**
Changelog
 

- [x] add validators 
- [x]  create models and its controllers => movie, client, genre
- [x] rework model => ticket

[ ] Concept
How to know if I can go to that movie? 
Each ticket is bought to specified cinema_hall(!), We know client is interested in movie that starts at 6:00pm
All tickets cost the same, though cost attribute is not needed here, just concept
Count All clients that have tickets to specified cinema_hall and are interested in movie that is being played.

### CODE REVIEW

**Comment by KMosc** [comment]
I would like to hear your opinions @wmaciejak @jandudulski

File: app/models/ticket.rb [hunk 1]
```diff
@@ -1,6 +1,9 @@
 class Ticket < ApplicationRecord
+  belongs_to :ticket_desk
   belongs_to :cinema_hall
+  belongs_to :client
   validate :id, :if => :exist?
```
Comment (wmaciejak) [inline app/models/ticket.rb:5]: Consider if this validation has a general sense? ID is an attribute which has to be generated during creation in DB

Comment (KMosc) [reply app/models/ticket.rb:5]: Well, I was thinking about caching id in session. We cant buy if we "log in", then we have id cached in session and if we buy, id of logged in user is passed as param.

File: app/models/client.rb [hunk 2]
```diff
@@ -0,0 +1,3 @@
+class Client < ApplicationRecord
```
Comment (jandudulski) [inline app/models/client.rb:1]: What is client?

File: spec/models/client_spec.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe Client, type: :model do
```
Comment (w0st) [inline spec/models/client_spec.rb:3]: I prefer to delete files like that if you don't need to test any methods of `Client` class.If you decide to do it later -> you can always create the file then.

File: app/models/client.rb (hunk 2 above)
Comment (KMosc) [reply app/models/client.rb:1]: owner of the ticket

Comment (jandudulski) [reply app/models/client.rb:1]: You have `belongs_to :movie` relation here. I believe that: 1) the relation should be with a ticket, not movie ;) 2) `belongs_to` would mean that client can have just a single ticket - I think it should be the other way around

Comment (KMosc) [reply app/models/client.rb:1]: True.

**Pull request merged**